lib/presentation/core/widgets/authentication/form/selection/gender_dropdown.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import '../../../../../../utils/extensions.dart';
+import 'custom_dropdown_button_form_field.dart';
+
+class GenderDropdown extends StatefulWidget {
+  const GenderDropdown({super.key});
+  @override
+  State<GenderDropdown> createState() => _GenderDropdownState();
+}
+
+class _GenderDropdownState extends State<GenderDropdown> {
+  String? selected;
+  @override
+  Widget build(BuildContext context) {
+    selected ??= context.l10n.genderMale;
+    return CustomDropdownButtonFormField(
+      items: [
+        context.l10n.genderMale,
+        context.l10n.genderFemale,
+        context.l10n.genderOther
+      ],
+      prefixIcon: selected == context.l10n.genderMale
```

Comment:
Nested `? :` shorthand for conditionals isn't easy to read, so I think it should be avoided, especially in `build` function. Separating it to another widget would require a little bit more code, but it would definitely be more readable.

```dart
Widget genderPrefixIcon(BuildContext context, String? current) {
  if (current == context.l10n.genderMale) {
    return const Icon(Icons.male);
  }
  if (current == context.l10n.genderFemale) {
    return const Icon(Icons.female);
  }

  return const Icon(Icons.transgender);
}
```